src/base/NekRSProblem.C

```diff
@@ -109,9 +112,15 @@ NekRSProblem::NekRSProblem(const InputParameters & params)
     indices.mesh_velocity_x = start++ * nekrs::scalarFieldOffset();
     indices.mesh_velocity_y = start++ * nekrs::scalarFieldOffset();
     indices.mesh_velocity_z = start++ * nekrs::scalarFieldOffset();
+    indices.filtered_velocity_x = start++ * nekrs::scalarFieldOffset();
```

Comment:
Hi David, would you please elaborate upon what these filtered velocity components are? Would someone want both the `mesh_velocity_x` and `filtered_velocity_x`, or does this instead represent some option we should have for computing `mesh_velocity_x`?

Reply:
Sure, from my understanding these are important if there is any compression in the solid mesh, as with an incompressible flow solver this will cause huge instabilities. The filtered velocity calculates the integral of mesh_v dotted with the surface normal, and then subtracts this from mesh_v (so filtered_v = mesh_v - integral(mesh_v dot n). We apply the filtered v to the velocity boundary condition in Nek and then apply the unfiltered mesh velocity to the mesh boundary condition in Nek. For the cases I've run so far, this the integral term is really small (like 1e-12 or something), but Dr. Merzari has said that it is important for other FSI problems that they plan on doing.

Reply:
Thanks! I'm going to add a commit on top of this branch then, some changes need to happen in order to pass the existing test suite.

What do you think of making the filtered velocity computation an optional setting, which maybe defaults to off? This will reduce the amount of memory we need to allocate in the scratch space all the time, and may be useful for FSI problems to test turning it on/off.

Reply:
Sure, I can make it optional!

Reply:
Thanks! Once you do that, I'll add a commit on top of yours. Please ping me once the option has been added.